forc-pkg/src/pkg.rs

```diff
@@ -2161,6 +2161,9 @@ pub fn build_with_options(build_options: BuildOpts) -> Result<Built> {
     };
 
     let outputs = member_filter.filter_outputs(&build_plan, outputs);
+    if outputs.is_empty() {
+        bail!("Nothing to build after filter");
```

Comment:
It would be better to define error types for `forc-pkg`. Matching on the string is not very maintainable because someone could easily change the string without updating the code that checks for this string. In `sway-lsp`  we use [thiserror](https://docs.rs/thiserror/latest/thiserror/)

Reply:
Yeah that is something i want so bad for quite some time but never got the chance, I think @gr00vytvniks might be working on it rn. cc @eureka-cpu 

You can see #4817, #1201 and #3283

Reply:
Let's add TODO comments with the issue number here and in the places where the string is checked, so don't lose sight of it.

Reply:
I am converting this one to draft as I already started working on #4817 and will try to merge that PR first